osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/networking/Mx382AlarmDecoder.java

```diff
@@ -17,72 +17,104 @@ public class Mx382AlarmDecoder extends AlarmDecoder {
     this.addAlarm();
 
     return this.builder.build();
   }
 
-  private void skipDateTime(final InputStream inputStream) throws UnrecognizedMessageDataException {
-    // The first byte of the datetime is the length byte
-    final byte dateTimeLength = this.readByte(inputStream);
-    this.builder.appendByte(dateTimeLength);
-    // Skip the rest of the datetime bytes
-    final byte[] dateTimeBytes = this.readBytes(inputStream, dateTimeLength);
-    this.builder.appendBytes(dateTimeBytes);
+  private void handleDateTime(final InputStream inputStream)
+      throws UnrecognizedMessageDataException {
+    // The first byte tells if datetime is provided 0x00 is not provided and 0x09 is provided
+    final byte dataType = this.readByte(inputStream);
+
+    // read the rest of the datetime bytes date is provided
+    if (dataType == DATA_TYPE_OCTET_STRING) {
+      // The next byte of the datetime is the length byte
+      final byte lengthOfData = this.readByte(inputStream);
+      this.readBytes(inputStream, lengthOfData);
+    }
   }
 
-  private void decodeEquipmentIdentifier(final InputStream inputStream)
+  private void handleEquipmentIdentifier(final InputStream inputStream)
       throws UnrecognizedMessageDataException {
 
-    // Check bytes of incoming mx382 message
-    this.checkBytes(inputStream, DEVICE_ID_2_CLASS_ID, "class-id");
-    this.checkBytes(inputStream, AMM_FORWARDED_ALARM_VERSION_0, "logical name forwarded alarm");
-    this.checkByte(inputStream, VALUE_ATTRIBUTE_ID, "attribute-id");
+    final byte dataType = this.readByte(inputStream);
+    this.builder.appendByte(dataType);
+    final byte length = this.readByte(inputStream);
+    this.builder.appendByte(length);
 
     // Read the equipment identifier from incoming mx382 message
-    final byte[] equipmentIdentifierBytes =
-        this.readBytes(inputStream, DSMR22_EQUIPMENT_IDENTIFIER_LENGTH);
+    final byte[] equipmentIdentifierBytes = this.readBytes(inputStream, length);
     this.builder.withEquipmentIdentifier(
         new String(equipmentIdentifierBytes, StandardCharsets.US_ASCII));
   }
 
+  private void handleCosemAttributeDescriptor(final InputStream inputStream)
+      throws UnrecognizedMessageDataException {
+    // Check bytes of incoming mx382 message
+    this.checkBytes(inputStream, CLASS_ID_1, "class-id");
+    this.checkBytes(inputStream, AMM_FORWARDED_ALARM_VERSION_0, "logical name forwarded alarm");
+    this.handleByte(inputStream, ATTRIBUTE_ID_2, "attribute-id");
+  }
+
   private void addAlarm() {
     this.builder.withTriggerType(PUSH_ALARM_TRIGGER);
     final Set<AlarmTypeDto> alarmSet = new HashSet<>();
     alarmSet.add(AlarmTypeDto.LAST_GASP);
     this.builder.addAlarms(alarmSet);
   }
 
-  private void checkByte(final InputStream inputStream, final byte expectedByte, final String name)
+  private void handleByte(final InputStream inputStream, final byte expectedByte, final String name)
```

Comment:
Why is this renamed to handleByte? It seems to do the same as checkBytes, but then only for 1 byte.

Reply:
renamed back to checkBytes